File: backend/src/models/Friendship.ts
import mongoose, { Document, Schema } from 'mongoose';

export interface IFriendship extends Document {
  userOneId: mongoose.Types.ObjectId;
  userTwoId: mongoose.Types.ObjectId;
  createdAt: Date;
  updatedAt: Date;
}

const FriendshipSchema = new Schema<IFriendship>(
  {
    userOneId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
    userTwoId: {
      type: Schema.Types.ObjectId,
      ref: 'User',
      required: true,
    },
  },
  {
    timestamps: true,
  }
);

// Ensure duplicate friendships cannot exist
FriendshipSchema.index({ userOneId: 1, userTwoId: 1 }, { unique: true });

export const Friendship = mongoose.model<IFriendship>('Friendship', FriendshipSchema);

File: backend/app/models/friendship.py
from datetime import datetime
from pydantic import Field
from beanie import Document
from beanie import PydanticObjectId as ObjectId
import pymongo

class Friendship(Document):
    userOneId: ObjectId
    userTwoId: ObjectId
    createdAt: datetime = Field(default_factory=datetime.utcnow)
    updatedAt: datetime = Field(default_factory=datetime.utcnow)

    class Settings:
        name = "friendships"
        indexes = [
            pymongo.IndexModel([("userOneId", pymongo.ASCENDING), ("userTwoId", pymongo.ASCENDING)], unique=True)
        ]
